src/main/java/de/fearnixx/jeak/teamspeak/voice/connection/VoiceConnection.java

```diff
@@ -0,0 +1,223 @@
+package de.fearnixx.jeak.teamspeak.voice.connection;
+
+import com.github.manevolent.ts3j.command.CommandException;
+import com.github.manevolent.ts3j.event.TS3Listener;
+import com.github.manevolent.ts3j.event.TextMessageEvent;
+import com.github.manevolent.ts3j.protocol.socket.client.LocalTeamspeakClientSocket;
+import de.fearnixx.jeak.IBot;
+import de.fearnixx.jeak.event.query.QueryEvent;
+import de.fearnixx.jeak.service.event.IEventService;
+import de.fearnixx.jeak.service.teamspeak.IUserService;
+import de.fearnixx.jeak.teamspeak.EventCaptions;
+import de.fearnixx.jeak.teamspeak.PropertyKeys;
+import de.fearnixx.jeak.teamspeak.data.IClient;
+import de.fearnixx.jeak.teamspeak.voice.connection.event.VoiceConnectionTextMessageEvent;
+import de.fearnixx.jeak.teamspeak.voice.connection.info.AbstractVoiceConnectionInformation;
+import de.fearnixx.jeak.teamspeak.voice.sound.AudioPlayer;
+import de.fearnixx.jeak.teamspeak.voice.sound.Mp3AudioPlayer;
+import de.fearnixx.jeak.voice.connection.IVoiceConnection;
+import de.fearnixx.jeak.voice.connection.IVoiceConnectionInformation;
+import de.fearnixx.jeak.voice.sound.AudioType;
+import de.fearnixx.jeak.voice.sound.IAudioPlayer;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+public class VoiceConnection implements IVoiceConnection {
+
+    private final AbstractVoiceConnectionInformation clientConnectionInformation;
+    private final String hostname;
+    private final int port;
+    private final IEventService eventService;
+
+    private LocalTeamspeakClientSocket ts3jClientSocket;
+    private boolean locked;
+
+    private boolean connected;
+
+    private IUserService userService;
+
+    private IBot bot;
+
+    private boolean shouldForwardTextMessages;
+
+    VoiceConnection(AbstractVoiceConnectionInformation clientConnectionInformation, String hostname, int port, IEventService eventService, IBot bot, IUserService userService) {
+        this.clientConnectionInformation = clientConnectionInformation;
+        this.hostname = hostname;
+        this.port = port;
+        this.eventService = eventService;
+        this.userService = userService;
+        this.bot = bot;
+        locked = true;
+    }
+
+    @Override
+    public void connect(Runnable onSuccess, Runnable onError) {
+        if (connected) {
+            return;
+        }
+
+        new Thread(
```

Comment:
In accordance to Java 8+ conventions, please use an executor build from ``#newSingleThreadPoolExecutor``.